### PR TITLE
changes for audio-based avatar

### PR DESCRIPTION
**Pull request by shilcare**
1. extend video API to support audio-based avatar
2. update doc comments

### CODE REVIEW

File: packages/teams-js/src/public/video.ts [hunk 1]
```diff
@@ -93,32 +140,37 @@ export namespace video {
...
-   * video extension should call this to notify host client that the current selected effect parameter changed.
+   * Video extension should call this to notify host client that the current selected effect changed.
    * If it's pre-meeting, host client will call videoEffectCallback immediately then use the videoEffect.
-   * If it's the in-meeting scenario, we will call videoEffectCallback when apply button clicked.
+   * If it's the in-meeting scenario, host client will call videoEffectCallback when apply button clicked.
    *
-   * @param effectChangeType - the effect change type.
+   * @param effectChangeType - The effect change type.
    * @param effectId - Newly selected effect id.
+   * @param effectParam Parameter for the selected effect.
```
Comment (arocc) [inline packages/teams-js/src/public/video.ts:162]: Why is an additional param besides id needed? What would this string be used for?

Comment (shilcare) [reply packages/teams-js/src/public/video.ts:162]: It's used for avatar background

File: packages/teams-js/src/private/videoEx.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import { video } from '../public/video';
+
+/**
+ * @hidden
+ * Personalized video effect
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export interface PersonalizedEffect {
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/private/videoEx.ts:13]: Please add unit tests for the functions in this file

File: packages/teams-js/src/public/video.ts [hunk 3]
```diff
@@ -38,6 +38,14 @@ export namespace video {
      * RGB stride, valid only when video frame format is RGB
      */
     stride?: number;
+    /**
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/public/video.ts:41]: Please add unit tests for the functions in this file

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

File: packages/teams-js/src/private/videoEx.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import { sendMessageToParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from '../public/constants';
+import { video } from '../public/video';
+
```
[the remark this answers is not in the document]
Comment (shilcare) [reply packages/teams-js/src/private/videoEx.ts:6]: Added

File: change/@microsoft-teams-js-dc703958-89fd-473f-bbe2-1e33d2102e69.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Extending video API to support audio-driven avatar ",
```
[the remark this answers is not in the document]
Comment (shilcare) [reply change/@microsoft-teams-js-dc703958-89fd-473f-bbe2-1e33d2102e69.json]: updated

File: packages/teams-js/src/public/video.ts [hunk 6]
```diff
@@ -117,16 +145,18 @@ export namespace video {
    * @beta
    * @param effectChangeType - the effect change type.
    * @param effectId - Newly selected effect id.
+   * @param effectParam Parameter for the selected effect.
    */
   export function notifySelectedVideoEffectChanged(
     effectChangeType: EffectChangeType,
     effectId: string | undefined,
+    effectParam?: string,
```
[the remark this answers is not in the document]
Comment (shilcare) [reply packages/teams-js/src/public/video.ts]: moved this change to the private folder as well

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

File: packages/teams-js/src/private/videoEx.ts [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface VideoFrameConfig extends video.VideoFrameConfig {
+    /**
+     * @hidden
+     * Flag to indicate use camera stream to synthesize video frame or not.
+     * Default value is true.
+     *
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/private/videoEx.ts:29]: this should also have @beta like everything else

File: packages/teams-js/src/private/videoEx.ts [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+     * @hidden
+     * Machine learning model for audio inference
+     *
+     * @internal
+     * Limited to Microsoft-internal use
+     */
+    audioInferenceModel?: ArrayBuffer;
+  }
+
+  export type VideoFrameCallback = (
```
[the remark this answers is not in the document]
Comment (shilcare) [reply packages/teams-js/src/private/videoEx.ts:73]: The `VideoFrame` type is different

File: packages/teams-js/src/private/videoEx.ts (hunk 7 above)
Comment (TrevorJoelHarris) [reply packages/teams-js/src/private/videoEx.ts:29]: Please actually add the @beta tag

**Review by TrevorJoelHarris — APPROVED** [review]

**Review by TrevorJoelHarris — APPROVED** [review]

**Review by erinha — APPROVED** [review]

**Pull request merged**